src/test/programs/federal/snap/unit.test.js

```diff
@@ -535,13 +543,257 @@ describe('SNAPhelpers', () => {
+      const shelterCost = 1;
+      const adjustedIncome = 100;
+      getTotalshelterCost.mockReturnValue(shelterCost);
+      getHalfAdjustedIncome.mockReturnValue(adjustedIncome);
+
+      const rawDeduction = shelterCost - adjustedIncome;
+      expect(rawDeduction).toBeLessThan(0);
+      expect(SNAPhelpers.getRawShelterDeduction( current )).toEqual(0);
+    });
+  });
+
+
   // `SNAPhelpers.getShelterDeduction()`
+  describe('`.getShelterDeduction( timeClient )` given a time-restricted client object with', () => {
+    let current, getRawShelterDeduction;
+    beforeEach(() => {
+      current = cloneDeep( defaultCurrent );
+      getRawShelterDeduction = jest.spyOn(SNAPhelpers, 'getRawShelterDeduction');
+    });
+    afterEach(() => {
+      getRawShelterDeduction.mockRestore();
+    });
+
+    it('that has an elderly or disabled member, it should return the raw deduction', () => {
+      current.household.push({ m_age: 65, m_disabled: true, m_role: 'member' });
+      const rawDeduction = 12;
+      getRawShelterDeduction.mockReturnValue(rawDeduction);
+      expect(SNAPhelpers.getShelterDeduction( current )).toEqual(rawDeduction);
+    });
+
+    it('that has a raw deduction under the cap, it should return the raw deduction', () => {
+      const rawDeduction = SNAPData.SHELTER_DEDUCTION_CAP - 1;
+      getRawShelterDeduction.mockReturnValue(rawDeduction);
+      expect(SNAPhelpers.getShelterDeduction( current )).toEqual(rawDeduction);
+    });
+
+    it('that has a raw deduction above the cap, it should return the cap', () => {
+      const rawDeduction = SNAPData.SHELTER_DEDUCTION_CAP + 1;
+      getRawShelterDeduction.mockReturnValue(rawDeduction);
+      expect(SNAPhelpers.getShelterDeduction( current )).toEqual(SNAPData.SHELTER_DEDUCTION_CAP);
+    });
+  });
+
+
   // `SNAPhelpers.getHomelessDeduction()`
+  describe('`.getHomelessDeduction() given a time-restriced client object with', () => {
+    let current;
+    beforeEach(() => {
+      current = cloneDeep( defaultCurrent );
+    });
+
+    it('that is homeless, it should return the homeless deduction', () => {
+      current.shelter = 'homeless';
+      expect(SNAPhelpers.getHomelessDeduction( current )).toEqual(SNAPData.HOMELESS_DEDUCTION);
+    });
+
+    it('that is not homeless, it should return zero', () => {
+      current.shelter = 'homeowner';
+      expect(SNAPhelpers.getHomelessDeduction( current )).toEqual(0);
+    });
+  });
+
+
   // `SNAPhelpers.getAdjustedNotGrossIncome()`
+  describe('`.getAdjustedNotGrossIncome( timeClient )` given a time-restricted client object with', () => {
+    let current, mocks,
+      adjustedGross, standardDeduction, earnedIncomeDeduction, medicalDeduction, dependentCareDeduction;
+    beforeEach(() => {
+      current = cloneDeep( defaultCurrent );
+      mocks = [];
+    });
+    afterEach(() => {
+      mocks.forEach(mock => mock.mockRestore());
+    });
+
+    const mockHelpers = () => {
```

Comment:
May be more clear if we just pass four arguments into here instead of playing around with how scope works in jest.

Reply:
Or maybe two values considering what's below.